va/validation-authority.go

```diff
@@ -272,6 +279,20 @@ func (va *ValidationAuthorityImpl) fetchHTTP(ctx context.Context, identifier cor
 			Detail: fmt.Sprintf("Error reading HTTP response body: %v", err),
 		}
 	}
+	// io.LimitedReader will silently truncate a Reader so if the
+	// resulting payload is the same size as maxResponseSize fail
+	if len(body) >= maxResponseSize {
+		return nil, validationRecords, &probs.ProblemDetails{
+			Type: probs.UnauthorizedProblem,
+			Detail: fmt.Sprintf(
+				"Invalid response from %s [%s]: response was >= %d bytes), body=\"%s\"",
```

Comment:
I think telling the user what `maxResponseSize` is is not necessary, and at best confusing. By far the most common case will be that they're just returning the wrong thing, in which case it will be immediately obvious what the problem is.  Also, the IP address is available in ValidationRecord, no need to duplicate it here. How about:

```
"Invalid response from %s: \"%s\"", url.String(), body
```